### PR TITLE
[codex] Implement web-intl i18n backend and adapter helpers

### LINKED ISSUE #124: task: Implement web-intl NumberFormatter backend in ars-i18n
**Wave 4-5** | **Points: 5** | **Epic: #54 (I18n)**

## Goal

Implement the `web-intl` backend for `ars-i18n::NumberFormatter` so WASM client builds can use the browser's `Intl.NumberFormat` API instead of ICU4X compiled data.

## Layer

Subsystem

## Framework

None

## Test tier

Unit

## Depends on

- #75 (ICU4X Locale integration)
- #79 (ICU4X-backed NumberFormatter foundation)

## Spec refs

- `spec/foundation/04-internationalization.md`
- `spec/testing/14-ci.md`

## Files to create/modify

- `crates/ars-i18n/Cargo.toml`
- `crates/ars-i18n/src/lib.rs`
- `crates/ars-i18n/src/number.rs` or `crates/ars-i18n/src/web_intl/number.rs`
- `spec/foundation/04-internationalization.md`
- `spec/testing/14-ci.md` only if checks/spec drift need sync

## Tests to add first

- Feature-gated tests or compile coverage proving `ars-i18n` builds with `--no-default-features --features web-intl --target wasm32-unknown-unknown`
- Tests for option mapping of decimal, percent, and currency formatting
- Tests proving `icu4x` and `web-intl` are mutually exclusive
- Tests or smoke coverage for locale separator extraction/parsing behavior under `web-intl`

## Acceptance criteria

- `web-intl` is a real backend, not an empty feature flag.
- `NumberFormatter` keeps the same public API under both `icu4x` and `web-intl`.
- Browser builds use `Intl.NumberFormat` for decimal, percent, and currency formatting.
- `format_percent(0.47, None)` preserves ars-ui fractional semantics and formats as `47%`.
- `format_currency()` preserves ISO-4217 minor-unit defaults.
- `decimal_separator()` and `grouping_separator()` work under `web-intl`.
- `parse()` remains locale-aware for browser-backed formatting.
- `icu4x` and `web-intl` cannot be enabled together.
- CI/verification includes the wasm `web-intl` cargo check path required by the spec.

## Spec impact

Potentially yes.

The current spec still sketches backend-specific public types (`JsIntlNumberFormatter`, `DefaultNumberFormatter`, `NumberFormat` trait). If implementation keeps the concrete public `NumberFormatter` API and uses backend separation only internally, update the spec in the same task to remove stale public backend wrappers.

### LINKED ISSUE #140: task: Implement t() function for Leptos and Dioxus adapters
**Wave 5a** | **Points: 3** | **Epic: #54 (I18n)**

## Goal

Implement the adapter-specific `t()` function that bridges `Translate` enums to reactive rendering in both framework adapters.

## Layer

Adapter

## Framework

Leptos, Dioxus

## Test tier

Unit

## Depends on

- #75 (ICU4X Locale integration)
- #137 (Translate trait)
- Adapter ArsContext infrastructure

## Spec refs

- `spec/foundation/04-internationalization.md` §7.5, §10.1, §10.2

## Scope

- **Leptos** `t()`: reads locale/icu from `ArsProvider`, returns reactive closure subscribing to locale signal (fine-grained reactivity)
- **Dioxus** `t()`: reads locale/icu from `ArsProvider`, returns `String` via `Signal::read()` (component-level reactivity)
- Both exported in adapter preludes
- Fallback to en-US + `StubIcuProvider` when no `ArsProvider` present

## Tests to add first

- `t(MyEnum::Title)` produces correct text for current locale
- Locale change causes re-evaluation
- Falls back to en-US + StubIcuProvider without ArsProvider
- Exported in `ars_leptos::prelude` and `ars_dioxus::prelude`

## Acceptance criteria

- Leptos `t()` returns reactive closure per spec §7.5
- Dioxus `t()` returns String per spec §7.5
- Both follow spec §10.1 and §10.2 signatures
- Falls back gracefully without ArsProvider
- Re-exported in both preludes
- All tests TDD-first, zero warnings

### LINKED ISSUE #143: task: Implement WebIntlProvider for IcuProvider trait in ars-i18n
**Wave 5b** | **Points: 5** | **Epic: #54 (I18n)**

## Goal

Implement a browser-backed IcuProvider using Intl APIs for WASM client builds, providing full parity with Icu4xProvider.

## Layer

Subsystem

## Framework

None

## Test tier

Unit

## Depends on

- #75 (ICU4X Locale integration)
- #138 (IcuProvider full trait definition)

## Spec refs

- `spec/foundation/04-internationalization.md` §9.5.4

## Scope

- Implement `WebIntlProvider` behind `#[cfg(feature = "web-intl")]`
- All 11 `IcuProvider` methods using browser APIs per spec §9.5.4
- Wire `default_provider()` web-intl arm to return `WebIntlProvider`

### Browser API mapping

| Method | Browser API | Strategy |
|---|---|---|
| `weekday_short_label` | `Intl.DateTimeFormat(locale, {weekday:'short'}).format(date)` | Clean |
| `weekday_long_label` | `Intl.DateTimeFormat(locale, {weekday:'long'}).format(date)` | Clean |
| `month_long_name` | `Intl.DateTimeFormat(locale, {month:'long'}).format(date)` | Clean |
| `day_period_label` | `Intl.DateTimeFormat` formatToParts dayPeriod | Clean |
| `day_period_from_char` | Parse against cached AM/PM labels | Lookup |
| `format_segment_digits` | `Intl.NumberFormat(locale, {minimumIntegerDigits})` | Clean |
| `max_months_in_year` | Lookup table + date probing | Lookup |
| `days_in_month` | Date arithmetic (Gregorian direct; others via probing) | Lookup |
| `hour_cycle` | `Intl.DateTimeFormat().resolvedOptions().hourCycle` | Clean |
| `first_day_of_week` | `Intl.Locale().getWeekInfo().firstDay` + fallback table | Clean+table |
| `convert_date` | `Intl.DateTimeFormat` with target calendar + reparse | Format+parse |

## Tests to add first

- All 11 methods return correct results for en-US, ar-SA, ja-JP
- `weekday_short_label(Monday, &en)` returns a short English weekday
- `hour_cycle(&en_us)` returns H12
- `first_day_of_week(&us)` returns Sunday
- `default_provider()` returns `WebIntlProvider` under `web-intl`

## Acceptance criteria

- All 11 `IcuProvider` methods implemented under `web-intl`
- Methods with clean browser mappings produce locale-correct output
- Calendar-specific methods have documented fallback strategies per spec §9.5.4
- `default_provider()` returns `WebIntlProvider` under `web-intl`
- Builds with `--no-default-features --features web-intl --target wasm32-unknown-unknown`
- All tests TDD-first, zero warnings

### PR DESCRIPTION
**Pull request by fogodev**
## Summary

This PR adds the browser-backed `web-intl` number formatting backend in `ars-i18n`, wires the `WebIntlProvider`/ambient number formatter access through both adapters, and syncs the affected specs and CI feature-matrix coverage.

## What Changed

- add the wasm `web-intl` `NumberFormatter` backend with dedicated unit/browser coverage
- add adapter-side `use_number_formatter(...)` helpers and finish `t()`/prelude wiring for Leptos and Dioxus
- update `cargo xci` / CI feature-matrix coverage and the related spec sections to match the shipped behavior
- remove a few test helper warning sites uncovered while validating the branch end-to-end

## Impact

WASM client builds can now use browser `Intl.NumberFormat` without changing the public `NumberFormatter` API, and both adapters expose reactive access to locale-aware number formatting and translated application text from `ArsProvider`.

## Validation

- `cargo xci`

Closes #124
Closes #140
Closes #143

### CODE REVIEW

File: crates/ars-i18n/src/number.rs [hunk 1]
```diff
@@ -696,42 +796,66 @@ fn round_half_even(value: f64) -> f64 {
...
+fn fallback_format(value: f64, formatter: &NumberFormatter) -> String {
+    let precision = usize::from(formatter.options.max_fraction_digits);
+
+    let rounded = formatter.prepare_numeric_value(value, true);
+
+    let mut output = if precision == 0 {
+        format!("{rounded:.0}")
+    } else {
+        format!("{rounded:.precision$}")
+    };
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number.rs]: Fixed in `ec58665`.

The fallback formatter now reuses the same fixed-decimal preparation path as the ICU-backed formatter, so `min_integer_digits`, `min_fraction_digits`, and `sign_display` are preserved instead of forcing `max_fraction_digits` precision in the fallback path. I also added fallback-only tests covering digit padding and negative-zero sign behavior.

Validation: `cargo xci` passed, including coverage (`ars-i18n` 96.6% lines / 87.6% branches).

File: crates/ars-i18n/src/number/web_intl.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            js_options.set_currency_sign(CurrencySign::Standard);
+        }
+
+        NumberStyle::Unit(unit) => {
+            let unit_id = resolve_measure_unit_id(unit)
+                .expect("unit formatter requires a resolvable CLDR unit id");
+
+            js_options.set_style(NumberFormatStyle::Unit);
+
+            js_options.set_unit(&unit_id);
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs]: Fixed locally.

`WebIntlNumberFormatter` now checks `Intl.supportedValuesOf("unit")` before setting `style: "unit"` / `unit`, and falls back to decimal formatting when the browser does not sanction the resolved CLDR unit ID. That keeps valid `MeasureUnit` values from crashing wasm apps during `Intl.NumberFormat` construction.

I also added a browser-backed wasm test that finds a CLDR unit unsupported by the current browser and asserts the formatter falls back to decimal output instead of throwing.

Validation: `CHROME="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome" wasm-pack test --headless --chrome crates/ars-i18n --no-default-features --features std,web-intl -- --nocapture web_intl_`

File: crates/ars-i18n/src/number/web_intl.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+fn build_options(options: &NumberFormatOptions) -> JsNumberFormatOptions {
+    let js_options = JsNumberFormatOptions::new();
+
+    js_options.set_minimum_integer_digits(options.min_integer_digits.get());
+
+    js_options.set_minimum_fraction_digits(options.min_fraction_digits);
+
+    js_options.set_maximum_fraction_digits(options.max_fraction_digits);
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs]: Fixed locally.

The `web-intl` option builder now normalizes fraction digit bounds before constructing `Intl.NumberFormat` by clamping `maximumFractionDigits` up to `minimumFractionDigits` when the caller provides an inverted range. That preserves the non-crashing behavior of the other backends instead of surfacing a browser `RangeError` on wasm.

I added a wasm test covering `min_fraction_digits: 3` with `max_fraction_digits: 1` and verified it formats as `1.200` rather than panicking.

Validation: `CHROME="/Applications/Google Chrome.app/Contents/MacOS/Google Chrome" wasm-pack test --headless --chrome crates/ars-i18n --no-default-features --features std,web-intl -- --nocapture web_intl_`

**Comment by fogodev** [comment]
@codex

File: crates/ars-i18n/src/number/web_intl.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs]: Fixed in `d5de43f`.

`WebIntlNumberFormatter` now guards `set_unit(...)` behind `Intl.supportedValuesOf("unit")` and falls back to decimal formatting when the resolved CLDR unit ID is not sanctioned by the browser. I kept the browser-backed wasm regression test that probes for an unsupported CLDR unit and verifies the formatter degrades gracefully instead of throwing.

While rerunning the full gate I also fixed a missing `alloc::string::String` import in the new wasm test helper so the coverage-targeted `wasm32` build compiles cleanly.

Validation:
- `cargo llvm-cov test -p ars-i18n` -> 96.34% lines / 96.57% regions locally
- `cargo xci`

File: crates/ars-i18n/src/number/web_intl.rs [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+fn normalized_fraction_digit_bounds(options: &NumberFormatOptions) -> (u8, u8) {
+    (
+        options.min_fraction_digits,
+        options.max_fraction_digits.max(options.min_fraction_digits),
+    )
+}
+
+fn browser_supports_unit(unit_id: &str) -> bool {
+    supported_values_of(SupportedValuesKey::Unit)
+        .iter()
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs:217]: Fixed in `84d8908`.

`browser_supports_unit` now feature-detects `Intl.supportedValuesOf`, verifies it is callable, and treats missing or throwing probes as unsupported instead of letting formatter construction panic. I also added a browser-backed wasm regression that temporarily removes `Intl.supportedValuesOf` and verifies unit formatting degrades to decimal output.

Validation:
- `wasm-pack test --headless --chrome crates/ars-i18n --no-default-features --features std,web-intl -- --nocapture web_intl_`
- `cargo llvm-cov test -p ars-i18n` -> 96.34% lines / 96.57% regions
- `cargo xci`

File: crates/ars-dioxus/src/provider.rs [hunk 5]
```diff
@@ -327,12 +336,44 @@ pub(crate) fn resolve_locale(adapter_props_locale: Option<&Locale>) -> Locale {
...
+{
+    let explicit_locale = adapter_props_locale.cloned();
+
+    let locale = use_locale();
+
+    use_memo(move || {
+        let resolved_locale = explicit_locale
+            .clone()
+            .unwrap_or_else(|| locale.read().clone());
+
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-dioxus/src/provider.rs]: Fixed in `84d8908`.

`use_resolved_number_formatter` now wraps the explicit locale and resolved `NumberFormatOptions` in `use_reactive!(...)` before memoizing, so prop-driven formatter inputs invalidate correctly instead of leaving a stale `NumberFormatter`. I added regressions covering both explicit-locale changes and non-reactive options changes across rerenders.

Validation:
- `cargo test -p ars-dioxus use_number_formatter_ -- --nocapture`
- `cargo test -p ars-dioxus use_resolved_number_formatter_ -- --nocapture`
- `cargo llvm-cov test -p ars-dioxus` -> 93.54% lines / 95.46% regions
- `cargo xci`

File: crates/ars-i18n/src/number/web_intl.rs [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+fn build_options(options: &NumberFormatOptions) -> JsNumberFormatOptions {
+    let js_options = JsNumberFormatOptions::new();
+
+    let (min_fraction_digits, max_fraction_digits) = normalized_fraction_digit_bounds(options);
+
+    js_options.set_minimum_integer_digits(options.min_integer_digits.get());
+
+    js_options.set_minimum_fraction_digits(min_fraction_digits);
+
+    js_options.set_maximum_fraction_digits(max_fraction_digits);
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs]: Clamped the JS-facing digit bounds before we build `Intl.NumberFormat` options. `minimumIntegerDigits` now clamps to `1..=21`, and the normalized fraction-digit bounds clamp to `0..=100`, so oversized Rust-side values no longer panic formatter construction. I also added the wasm regression `web_intl_clamps_out_of_range_digit_bounds_before_formatting` to lock the behavior down. Revalidated with `wasm-pack test --headless --chrome crates/ars-i18n --no-default-features --features std,web-intl -- --nocapture web_intl_`, `cargo llvm-cov test -p ars-i18n`, and `cargo xci`.

File: crates/ars-i18n/src/number/web_intl.rs [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+        js_options.set_sign_display(sign_display);
+    } else {
+        // `js-sys` does not expose the Intl `"negative"` variant yet, so
+        // set it through the raw options bag to preserve the spec-level
+        // distinction from `"auto"` for cases like negative zero.
+        Reflect::set(
+            js_options.as_ref(),
+            &JsValue::from_str("signDisplay"),
+            &JsValue::from_str("negative"),
+        )
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs:139]: Added capability detection before writing raw `signDisplay: "negative"`. If the browser rejects that constructor option, we now leave `signDisplay` unset and fall back safely instead of panicking formatter construction. I covered the rejection path with the wasm regression `web_intl_negative_sign_display_falls_back_when_browser_rejects_it`, using a test-only support override so the test stays narrow and doesn’t monkeypatch `Intl.NumberFormat` globally. Revalidated with `wasm-pack test --headless --chrome crates/ars-i18n --no-default-features --features std,web-intl -- --nocapture web_intl_`, `cargo llvm-cov test -p ars-i18n`, and `cargo xci`.

File: crates/ars-i18n/src/number/web_intl.rs [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+
+    /// Format the given numeric value through the browser formatter.
+    pub(super) fn format(&self, value: f64) -> String {
+        let format_fn: Function = self.inner.format();
+
+        format_fn
+            .call1(&JsValue::UNDEFINED, &JsValue::from_f64(value))
+            .ok()
+            .and_then(|value| value.as_string())
+            .unwrap_or_default()
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-i18n/src/number/web_intl.rs:53]: Fixed in `3f18d4e`.

I took the parser-side option here so locale-native `Intl.NumberFormat` output stays intact while negative round-trips work across backends. The shared number parser now accepts `U+2212` as a leading sign and normalizes it to ASCII `-` before parsing, and I added a regression in `number.rs` plus a browser-backed wasm regression `web_intl_parse_accepts_unicode_minus_from_browser_shaped_output` using `sv-SE` locale-shaped output.

Validation:
- `cargo test -p ars-i18n parses_signed_and_grouped_numbers_with_locale_heuristics -- --nocapture`
- `wasm-pack test --headless --chrome crates/ars-i18n --no-default-features --features std,web-intl -- --nocapture web_intl_`
- `cargo xci`
